### DISCUSSION TITLE
waitForElement + console.log() freezing the page

### DISCUSSION THREAD

**Discussion by TheColaber** [ScratchAddons/ScratchAddons#566] [Help] [answered]
```javascript
export default async function ({ addon, global, console }) {
  (async function () {
    while (true) {
      // ".gui_menu-bar-position_3U1T0" is the query selector for that top bar in the editor.
      const element = await addon.tab.waitForElement(".gui_menu-bar-position_3U1T0", { markAsSeen: true });
      console.log(element);
    }
  })();
}

```
But every time it runs, it freezes up the page. I don't know why this is happening...

Edit: #566 🎉 lol

**Comment by WorldLanguages** [comment]
What page is it freezing? After doing what actions?

**Reply by TheColaber** [reply]
@WorldLanguages It is freezing The editor/project (as it only ruins those scripts on those pages) page after running that script. In the console, it doesn't log anything.

**Reply by WorldLanguages** [reply]
Can you provide a screen recording?

**Reply by TheColaber** [reply]
https://www.youtube.com/watch?v=5C4Z4tIGj8E&feature=youtu.be

**Reply by WorldLanguages** [reply]
Hmm, are you sure it's not just taking a while to open the editor? You didn't wait that much. I cannot reproduce the issue with the same code, also Chrome. Going inside always freezes the page for me, with or without Scratch Addons enabled, I just need to wait a few seconds, depending on how busy my system is.

**Reply by TheColaber** [reply]
I did test this, I left it open for a minute, then this came up:
![image](https://user-images.githubusercontent.com/72760579/97116638-d8bbdb80-16d4-11eb-936c-6d0052d8965e.png)

**Reply by TheColaber** [reply]
Also, the project only has one short line of code, and without the addon enabled, it loads in like 2 seconds.

**Reply by WorldLanguages** [reply]
Can you disable all other addons?
Can you `console.log(addon.tab._waitForElementSet)` as the first line inside the `while(true)`?
And send another recording. Or if you stopped reproducing this after disabling other addons, tell me what addons you had enabled.

**Reply by TheColaber** [reply]
https://www.youtube.com/watch?v=pz-hmilqbOU&feature=youtu.be

**Reply by WorldLanguages** [reply]
Huh. That's weird...
Can you reproduce in Firefox? You don't have to record it, just tell me if it happens there as well, with other addons disabled.

**Reply by WorldLanguages** [reply]
Can you try not using `{waitForElement: true}`? Instead, set it to false, add a class to the element to mark it as seen, like this: `element.classList.add("seen")`. Then in the waitForElement call, use the same selector you use now but with `:not(.seen)` after it.

**Reply by TheColaber** [reply]
Hmm, Interesting, I simply removed the `console.log()` and it did the trick. I added a line of code like this: `element.classList.add("Thiswasadded")` and it worked... that's a weird bug. Do you know what is causing this?

**Reply by WorldLanguages** [reply]
Can you send the new code? I'm not sure what you mean. Solely removing the console.log made it work?

**Reply by TheColaber** [reply]
Nevermind...

**Reply by WorldLanguages** [reply]
A `console.log` breaking the page huh... Can you try renaming `console` at the top of the file, with `_console`? And see if that fixes it?

**Reply by TheColaber** [reply]
And also use `_console` down below in the code right?

**Reply by WorldLanguages** [reply]
No, that's the whole point. Only change it in the first line

**Reply by TheColaber** [reply]
Didn't work... even with

> ```js
> if (element) {
>   console.log("test")
> }
> ```

or 

> ```js
> if (element) {
>   _console.log("test")
> }
> ```

Edit: and yes I did change the top line to _console

**Reply by WorldLanguages** [reply]
Hmm, I think this might be caused because the Scratch editor for some reason replaces `console.log` with its own, and that's why all console logs appear to come from project.bundle.js

**Reply by TheColaber** [reply]
Yeah, that's true. I probably won't be using that anyway...

**Comment by WorldLanguages** [comment, marked as answer]
This issue was fixed by #622
